src/Microsoft.Extensions.SecretManager.Tools/Internal/ProjectIdResolver.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    public class ProjectIdResolver : IDisposable
+    {
+        private const string TargetsFileName = "FindUserSecretsProperty.targets";
+        private readonly ILogger _logger;
+        private readonly string _workingDirectory;
+        private Stack<Action> _disposables = new Stack<Action>();
```

Comment:
Why have an action that creates scopes when all you're doing is deleting created files?